discovery/kubernetes/endpointslice.go

```diff
@@ -145,8 +145,11 @@ func NewEndpointSlice(l *slog.Logger, eps cache.SharedIndexInformer, svc, pod, n
 				e.enqueueNode(node.Name)
 			},
 			DeleteFunc: func(o interface{}) {
-				node := o.(*apiv1.Node)
-				e.enqueueNode(node.Name)
+				nodeName, err := deletedNodeName(o)
```

Comment:
Hmm I'm not sure why we don't use the same pattern as for services where we call `serviceUpdate(interface{})` and let the update function check the object type?

Reply:
yes, but `serviceUpdate` is defined twice as it's not used in the same way, here we just need a util to get the node name... You know I think I'll go with https://github.com/prometheus/prometheus/pull/15170#discussion_r1802503869: enforce the keyFunc and just use `.Key`, it'll be one less cast.

Reply:
sharedInfromer doesn't allow to set `keyFunc`, so we'll assume it'll always be `MetaNamespaceKeyFunc`